### ISSUE TITLE
[Bug?] Suspicious warning plugin_typedtree/fully_qualified_identifiers/not-qualified-enough with objects

### ISSUE THREAD

**Issue by david-maison** [OCamlPro/typerex-lint#180]
With the following example
```
$ cat a.ml
class a = object method o = () end
let _ =
  object(self)
    inherit a
    method m = self#o
  end

$ cat .ocplint
plugin_parsing = {
  get_features = { enabled = false }
}
plugin_file_system = {
  interface_missing = { enabled = false }
}
plugin_parsetree = {
  check_class = { enabled = false }
}
```

With the following commands:
```
ocamlc -c -w +a-4 -bin-annot ./a.ml
ocp-lint --pwarning
```

It gives the following output:
```
Scanning files in project "."...
Found '8' file(s)
Running analyses... 7 / 7
Mergin database...
File "a.ml", line 5, characters 15-19:
  Warning 1: plugin_typedtree/fully_qualified_identifiers/not-qualified-enough SEVERITY 2
  external identifier "self" is not fully qualified
    (should be "self-2")
== New Warnings ==
  * 1 file(s) were linted
  * 1 warning(s) were emitted:
      * 1 warning(s) raised by "plugin_typedtree"/"fully_qualified_identifiers.1"/warning #1
  * 0 file(s) couldn't be linted
```

Is this warning expected?

**Comment by vprevosto** [comment]
Actually the example can be reduced further:

```ocaml
class a = object(self)
  method x = 1
  method y = self#x
end
```

will produce a `not-qualified-enough` warning for `self` in `self#y`
